server/docs/design/bidi-producer-consumers-streaming.md

```diff
@@ -0,0 +1,276 @@
+# Bi-directional Producer/Consumer Streaming with gRPC
+
+## Purpose
+
+A primary use case of the `hedera-block-node` is to stream live BlockItems (see Terms section) from a producer 
+(e.g. Consensus Node) to N consumers (e.g. Mirror Node) with the lowest possible latency while correctly preserving the 
+order of the BlockItems. This document outlines several possible strategies to implement this use case and the design 
+of the recommended approach. All strategies rely on the Helidon 4.x.x server implementations of HTTP/2 and gRPC 
+services to ingest BlockItem data from a producer and then to stream the same BlockItems to downstream consumers. It
+does this by defining bidirectional gRPC streaming services based on protobuf definitions.
```

Comment:
Good question.  Yes, the consumer-facing connection will be "server streaming" (unidirectional stream to the consumer after the initial request).  I've already implemented that in #29 but I thought I would adjust this design document with that PR so that the changes go together in the same PR.